st2debug/st2debug/cmd/submit_debug_info.py

```diff
@@ -345,38 +412,47 @@ def encrypt_archive(archive_file_path, debug=False):
     # pylint: disable=no-member
     assert import_result.count == 1
 
+    if config_yaml:
+	gpg_key_finger_print = get_config_details(config_yaml, 'gpg_key_fingerprint') 
+    else:
+	gpg_key_finger_print = GPG_KEY_FINGERPRINT
     encrypted_archive_output_file_path = archive_file_path + '.asc'
     with open(archive_file_path, 'rb') as fp:
         gpg.encrypt_file(fp,
-                         recipients=GPG_KEY_FINGERPRINT,
+                         recipients=gpg_key_finger_print,
                          always_trust=True,
                          output=encrypted_archive_output_file_path)
-
     return encrypted_archive_output_file_path
 
 
-def upload_archive(archive_file_path):
+def upload_archive(archive_file_path, config_yaml=False):
```

Comment:
Instead of passing in config_yaml should we pass in bucket_url=S3_BUCKET_URL and have the caller deal with parsing the YAML file.